scripts/travis_setup.sh

```diff
@@ -30,27 +30,27 @@ ${CXX} --version
 # Ensure mason is on the PATH
 export PATH="`pwd`/.mason:${PATH}" MASON_DIR="`pwd`/.mason"
 
-# Start the mock X server
-if [ -f /etc/init.d/xvfb ] ; then
-    mapbox_time "start_xvfb" \
-    sh -e /etc/init.d/xvfb start
-    sleep 2 # sometimes, xvfb takes some time to start up
-fi
-
-# Make sure we're connecting to xvfb
-export DISPLAY=:99.0
-
 mapbox_time "checkout_mason" \
 git submodule update --init .mason
 
 # Touch package.json so that we are definitely going to run an npm update action
 mapbox_time "touch_package_json" \
 touch package.json
 
-# Install and set up to load a more recent version of mesa
-mapbox_time "install_mesa" \
-mason install mesa 11.2.2
-export LD_LIBRARY_PATH="`mason prefix mesa 11.2.2`/lib:${LD_LIBRARY_PATH:-}"
+# Start the mock X server
+if [ -f /etc/init.d/xvfb ] && [ ! -z "${RUN_XVFB}" ]; then
+    mapbox_time "start_xvfb" \
+    sh -e /etc/init.d/xvfb start
+    sleep 2 # sometimes, xvfb takes some time to start up
+
+    # Make sure we're connecting to xvfb
+    export DISPLAY=:99.0
+
+    # Install and set up to load a more recent version of mesa
+    mapbox_time "install_mesa" \
+    mason install mesa 11.2.2
```

Comment:
BTW, I'm planning on a posterior PR to rework our Travis workflow e.g. the clang-tidy check shouldn't need gcc nor LLVM libs. Having a GLX-based mesa 13.0.0 could be part of that.